src/start/RealmScreen.js

```diff
@@ -17,7 +17,7 @@ type SelectorProps = {|
 
 type Props = $ReadOnly<{|
   navigation: NavigationScreenProp<{
-    params: ?{|
+    params?: {|
```

Comment:
Why optional at all? I think the logic in your previous version (at 6f6a3ee4a) was sound for why it shouldn't simply go missing; and following that logic we make `params` non-optional on other screens too.

Reply:
Ah I see, you answered that here :slightly_smiling_face: : https://github.com/zulip/zulip-mobile/pull/4273#discussion_r499095804

I think this calls for a comment saying why it's optional, in any case -- since that is out of the ordinary compared to how all, I think, of our other screens work.

... Then I went reading this code some more to see how it handles the case where `params` is missing. OK, effectively `realm` gets defaulted to `''`, but `initial`? It seems to default to `false`. So when is it ever true -- is that just dead code? ... No, it's not. The answer is this hack in `navReducer.js`:
```js
export const getStateForRoute = (route: string): NavigationState => {
  // TODO: this is kind of a hack!  Refactor to a better way.
  //  * Perhaps pass `initial: true` unconditionally in this initialization
  //    code, to all routes?  Then that'd just be part of the interface of
  //    making a screen work as an initial screen.
  //  * Alternatively, we could replace the whole system of `ModalNavBar`,
  //    `canGoBack`, etc., with our own "navigation view" that would be more
  //    directly integrated into the navigation framework:
  //      https://reactnavigation.org/docs/en/2.x/navigation-views.html
  const params = route === 'realm' ? { initial: true } : undefined;
```

So in fact in `navReducer` when we navigate to this screen, we *also* pass a `params` object (which I guess means we always do.) It has a property `initial: true`, and no `params`.

I guess now this extra definitely calls for a comment here :stuck_out_tongue: Which really I should have added at the same time as adding that hack: 7a04a5638

Reply:
Yeah, I've already got plans to get rid of that hack soon, as part of the process :)